tests/integration/test_code.py

```diff
@@ -0,0 +1,471 @@
+    get_hostname_to_use_for_compute_node,
+    removesuffix,
+)
+from milatools.utils import disk_quota
+from milatools.utils.compute_node import (
+    ComputeNode,
+    get_queued_milatools_job_ids,
+    sbatch,
+)
+from milatools.utils.disk_quota import check_disk_quota, check_disk_quota_v1
+from milatools.utils.remote_v1 import RemoteV1
+from milatools.utils.remote_v2 import RemoteV2
+
+from ..conftest import job_name, launches_jobs
+from .test_slurm_remote import get_recent_jobs_info_dicts
+
+logger = get_logger(__name__)
+
+
+async def _get_job_info(
+    job_id: int,
+    login_node: RemoteV2,
+    fields: tuple[str, ...] = ("JobID", "JobName", "Node", "WorkDir", "State"),
+) -> dict:
+    return dict(
+        zip(
+            fields,
+            (
+                await login_node.get_output_async(
+                    f"sacct --noheader --allocations --user=$USER --jobs {job_id} "
+                    "--format=" + ",".join(f"{field}%40" for field in fields),
+                    display=False,
+                    hide=True,
+                )
+            )
+            .strip()
+            .split(),
+        )
+    )
+
+
+@launches_jobs
+@pytest.mark.slow
+@pytest.mark.asyncio
+@pytest.mark.parametrize("persist", [True, False], ids=["sbatch", "salloc"])
+@pytest.mark.parametrize(
+    job_name.__name__,
+    # Don't set the `--job-name` in the `allocation_flags` fixture
+    # (this is necessary for `mila code` to work properly).
+    [None],
+    ids=[""],
+    indirect=True,
+)
+async def test_code(
+    login_node_v2: RemoteV2,
+    persist: bool,
+    capsys: pytest.CaptureFixture,
+    allocation_flags: list[str],
+    file_regression: FileRegressionFixture,
+    slurm_account_on_cluster: str,
+):
+    if login_node_v2.hostname == "localhost":
+        pytest.skip(
+            "TODO: This test doesn't yet work with the slurm cluster spun up in the GitHub CI."
+        )
+
+    home = await login_node_v2.get_output_async("echo $HOME")
+    scratch = await login_node_v2.get_output_async("echo $SCRATCH")
+
+    start = datetime.datetime.now() - timedelta(minutes=5)
+    jobs_before = get_recent_jobs_info_dicts(
+        login_node_v2, since=datetime.datetime.now() - start
+    )
+    jobs_before = {
+        int(job_info["JobID"]): job_info
+        for job_info in jobs_before
+        if job_info["JobName"] == "mila-code"
+    }
+
+    relative_path = "bob"
+
+    with contextlib.redirect_stderr(sys.stdout):
+        logger.info(f"{'sbatch' if persist else 'salloc'} flags: {allocation_flags}")
+        compute_node_or_job_id = await code(
+            path=relative_path,
+            command="echo",  # replace the usual `code` with `echo` for testing.
+            # idea: Could probably also return the process ID of the `code` editor?
+            persist=persist,
+            job=None,
+            node=None,
+            alloc=allocation_flags,
+            cluster=login_node_v2.hostname,  # type: ignore
+        )
```

Comment:
I think it's the case but just to be sure, are we using the job name `test-milatools` or something like that to identify test jobs or something like that and is that why we are sure to delete the remaining jobs even when we encounter errors?

Reply:
Not quite. When using `mila code`, we're preventing users from setting the job name flag, which also (at the moment) forces us to use the "real" `mila-code` job-name. I filter out my own jobs when scraping the recent jobs on the cluster. This is not ideal, and I'd be interested if you have suggestions on what we could do differently here, for sure.

As for the "we are sure to delete the remaining jobs even when we encounter errors", are you referring to the `cancel_new_jobs_on_interrupt`?